package NYT.NTransactionServer.NProto;

import "yt_proto/yt/core/misc/proto/guid.proto";

import "yt_proto/yt/core/ytree/proto/attributes.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqStartTransaction
{
    optional NYT.NYTree.NProto.TAttributeDictionary attributes = 1;
    optional string title = 6;

    optional NYT.NProto.TGuid parent_id = 3;
    repeated NYT.NProto.TGuid prerequisite_transaction_ids = 8;

    optional int64 timeout = 4;
    optional uint64 deadline = 10;

    required string user = 5;
    optional string user_tag = 12;

    repeated uint32 replicate_to_cell_tags = 9;

    optional bool is_cypress_transaction = 13;

    reserved 2, 11;
}

message TReqPrepareTransactionCommit
{
    required NYT.NProto.TGuid transaction_id = 1;
    required uint64 prepare_timestamp = 2;
    required string user = 3;
    optional string user_tag = 4;
}

// Used as a signal when replication of a missing transaction was requested.
message TReqNoteNoSuchTransaction
{
    required NYT.NProto.TGuid id = 1;
}

message TReqReturnBoomerang
{
    required NYT.NProto.TGuid boomerang_wave_id = 1;
    required int32 boomerang_wave_size = 2;

    required NYT.NProto.TGuid boomerang_mutation_id = 3;
    required string boomerang_mutation_type = 4;
    required bytes boomerang_mutation_data = 5;
}

message TReqRemoveStuckBoomerangWaves
{ }

////////////////////////////////////////////////////////////////////////////////
